### PR TITLE
En 3833 observers get out of mem

### PR DESCRIPTION
**Pull request by iulianpascalau**
Added a new implementation of DataPacker that calculates less precise but more optimum the packed size in bytes.

### CODE REVIEW

File: core/partitioning/simpleDataPacker.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+// SimpleDataPacker can split a large slice of byte slices in chunks <= maxPacketSize
+// If one element still exceeds maxPacketSize, it will be returned alone
+// It does the marshaling of the resulted (smaller) slice of byte slices
+type SimpleDataPacker struct {
+	marshalizer marshal.Marshalizer
+}
+
+// NewSimpleDataPacker creates a new SizeDataPacker instance
+func NewSimpleDataPacker(marshalizer marshal.Marshalizer) (*SimpleDataPacker, error) {
+	if marshalizer == nil || marshalizer.IsInterfaceNil() {
```
Comment (sasurobert) [inline core/partitioning/simpleDataPacker.go]: use your core func of isNil :)

File: core/partitioning/simpleDataPacker.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		isBuffToLarge := lenElements+len(element) >= limit
+		elementsNotEmpty := len(elements) > 0
+		if isBuffToLarge && elementsNotEmpty {
+			marshaledElements, _ := sdp.marshalizer.Marshal(elements)
+			returningBuff = append(returningBuff, marshaledElements)
+			elements = make([][]byte, 0)
+			lenElements = 0
+		}
+
+		elements = append(elements, element)
```
Comment (sasurobert) [inline core/partitioning/simpleDataPacker.go]: rename suggestion currentChunk

File: core/partitioning/simpleDataPacker.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
+package partitioning
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+)
+
+// SimpleDataPacker can split a large slice of byte slices in chunks <= maxPacketSize
```
Comment (sasurobert) [inline core/partitioning/simpleDataPacker.go]: you have the same commentary as sizeDataPacker - what is the difference - explain it in the comments. exaplin that it is not doing continuous marshalling. Add comment in the sizeDataPacker why is it doing continuous marshalling - precision, etc.

File: core/partitioning/simpleDataPacker.go (hunk 1 above)
Comment (iulianpascalau) [reply core/partitioning/simpleDataPacker.go]: Agree

Comment (iulianpascalau) [reply core/partitioning/simpleDataPacker.go]: Not available yet. It is on feat/interceptors branch.

File: core/partitioning/simpleDataPacker.go (hunk 2 above)
Comment (iulianpascalau) [reply core/partitioning/simpleDataPacker.go]: done

File: core/partitioning/simpleDataPacker.go (hunk 3 above)
Comment (iulianpascalau) [reply core/partitioning/simpleDataPacker.go]: added explanations

**Pull request merged**